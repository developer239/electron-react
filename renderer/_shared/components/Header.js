import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'


const HeaderContainer = styled.div`
  background-color: #03a9f4;
  width: 100%;
  padding: 28px 28px 10px 28px;
`

const HeaderText = styled.h1`
  color: white;
  font-weight: 400;
  line-height: 1.5;
  letter-spacing: 1.5px;
`

const Header = props => (
  <HeaderContainer>
    <HeaderText>{props.children}</HeaderText>
  </HeaderContainer>
)

Header.propTypes = {
  children: PropTypes.string.isRequired,
}

export default Header
