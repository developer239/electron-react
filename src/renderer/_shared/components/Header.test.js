import React from 'react'
import { shallow } from 'enzyme'
import Header from './Header'


describe('Header Component', () => {
  it('renders with all props', () => {
    const wrapper = shallow(<Header>My Header</Header>)
    expect(wrapper).toMatchSnapshot()
  })
})
