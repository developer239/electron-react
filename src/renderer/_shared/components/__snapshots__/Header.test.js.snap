// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component renders with all props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header>
    My Header
  </Header>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <styled.h1>
        My Header
      </styled.h1>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "My Header",
      },
      "ref": null,
      "rendered": "My Header",
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <styled.h1>
          My Header
        </styled.h1>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "My Header",
        },
        "ref": null,
        "rendered": "My Header",
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
